PythonAPI/docs/world.yml

```diff
@@ -0,0 +1,414 @@
+    - def_name: set_weather
+      params: 
+      - param_name: weather
+        type: carla.WeatherParameters
+      doc: >
+        Change the weather in the simulation.
+    # --------------------------------------
+    - def_name: get_snapshot
+      return: carla.WorldSnapshot
+      doc: >
+        Return a snapshot of the world at this moment.
+    # --------------------------------------
+    - def_name: get_actor
+      return: carla.Actor
+      params:
+      - param_name: actor_id
+        type: int
+      doc: >
+        Find actor by id, return nullptr if not found.
+    # --------------------------------------
+    - def_name: get_actors
+      return: carla.ActorList
+      doc: >
+        Return a list with all the actors currently present in the world.
+    # --------------------------------------
+    - def_name: get_actors
+      return: carla.ActorList
+      params: 
+      - param_name: actor_ids
+        type: list(int)
+      doc: >
+        Return a list with the actors requested by ActorId.
+    # --------------------------------------
+    - def_name: spawn_actor
+      return: carla.Actor
+      params:
+      - param_name: blueprint
+        type: carla.BlueprintLibrary
+        doc: >
+      - param_name: transform
+        type: carla.Transform
+        doc: >
+          If attached to parent, transform acts like a relative_transform to the parent actor.
+      - param_name: attach_to 
+        type: carla.Actor
+        default: None
+        doc: > 
+      - param_name: attachment 
+        type: carla.AttachmentType
+        default: Rigid
+        doc: > 
+      doc: >
+        Spawn an actor into the world based on the blueprint provided at
+        transform. If a parent is provided, the actor is attached to
+        parent.
+    # --------------------------------------
+    - def_name: try_spawn_actor
+      return: carla.Actor
+      params:
+      - param_name: blueprint
+        type: carla.BlueprintLibrary
+        doc: >
+      - param_name: transform
+        type: carla.Transform
+        doc: >
+          If attached to parent, transform acts like a relative_transform to the parent actor.
+      - param_name: attach_to 
+        type: carla.Actor
+        default: None
+        doc: > 
+      - param_name: attachment 
+        type: carla.AttachmentType
+        default: Rigid
+        doc: >
+      doc: >
+        Same as SpawnActor but return nullptr on failure instead of throwing an
+        exception.
+    # --------------------------------------
+    - def_name: wait_for_tick
+      return: carla.WorldSnapshot
+      params:
+      - param_name: seconds
+        type: float
+        default: 10.0
+      doc: >
+        Block calling thread until a world tick is received.
+    # --------------------------------------
+    - def_name: on_tick
+      params:
+      - param_name: callback
+      doc: >
+        Register a callback to be called every time a world tick is received.
```

Comment:
Return ID of the callback so it can be removed with `remove_on_tick`. The callback receives a WorldSnapshot as single argument.